uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/WorkPatternTest.php

```diff
@@ -502,5 +505,26 @@ public function testgetWorkDayTypeForDateShouldHaveCorrectDayTypeForPatternsWith
       new DateTime('2016-08-15'), $start
     ));
   }
-
+  public function testGetWorkDayType() {
+    //create a contact with valid contract and a Work pattern assigned
+    $contact = ContactFabricator::fabricate();
+    $periodStartDate = date('2016-01-01');
+
+    HRJobContractFabricator::fabricate([
+      'contact_id' => $contact['id']
+    ],
+    [
+      'period_start_date' => $periodStartDate,
+    ]);
+    $workPattern = WorkPatternFabricator::fabricateWithA40HourWorkWeek();
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $contact['id'],
+      'pattern_id' => $workPattern->id
+    ]);
+    $startDateTime = new DateTime('2016-11-30');
+
+    $type = WorKday::WORK_DAY_OPTION_YES;
+    $dayType = WorkPattern::getWorkDayType($contact['id'], $startDateTime);
+    $this->assertEquals($type, $dayType);
```

Comment:
This test is also a bit poor. Please add more variations with different scenarios